src/instructlab/model/full_train.py

```diff
@@ -220,16 +299,34 @@ def train(train_args, device, optimize_memory):
             for k in batch:
                 batch[k] = batch[k].to(device=dev)
 
-            output = model(**batch, use_cache=False, return_dict=True)
-            loss = output.loss
+            output = model(**batch, use_cache=False, return_dict=False)
+            loss = None
+            if isinstance(output, tuple):
+                loss = output[0]
+                assert (
```

Comment:
we should avoid asserts in code (there are different reasons for this, but one is: asserts can be disabled by runtime); if these were added for debugging purposes, then just remove them; if we think this is a legit scenario, then convert to `raise`s or handle the error case otherwise.